caom2-artifact-resolvers/src/main/java/ca/nrc/cadc/caom2/artifact/resolvers/MastResolver.java

```diff
@@ -0,0 +1,106 @@
+ * without limitation any war-          pris sans restriction toute
+ * ranty of merchantability or          garantie de valeur marchande
+ * fitness for a particular pur-        ou de pertinence pour un usage
+ * pose.  NRC shall not be liable       particulier.  Le CNRC ne
+ * in any event for any damages,        pourra en aucun cas etre tenu
+ * whether direct or indirect,          responsable de tout dommage,
+ * special or general, consequen-       direct ou indirect, particul-
+ * tial or incidental, arising          ier ou general, accessoire ou
+ * from the use of the software.        fortuit, resultant de l'utili-
+ *                                      sation du logiciel.
+ *
+ ****  C A N A D I A N   A S T R O N O M Y   D A T A   C E N T R E  *****
+ ************************************************************************
+ */
+
+package ca.nrc.cadc.caom2.artifact.resolvers;
+
+import ca.nrc.cadc.net.StorageResolver;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+import org.apache.log4j.Logger;
+
+/**
+ * This class can convert a MAST URI into a URL.
+ *
+ * @author jeevesh
+ */
+public class MastResolver implements StorageResolver {
+    public static final String SCHEME = "mast";
+    private static final Logger log = Logger.getLogger(MastResolver.class);
+    private static final String MAST_BASE_ARTIFACT_URL = "https://masttest.stsci.edu/partners/download/file";
+    private static final String CANNOT_GET_URL = "Can't generate URL from URI.";
+
+
+    public MastResolver() {
+    }
+
+    @Override
+    public String getScheme() {
+        return SCHEME;
+    }
+
+    @Override
+    public URL toURL(URI uri) {
+        this.validateScheme(uri);
+        String s = "";
+        try {
+            s = this.createURL(uri);
+            URL url = null;
+            url = new URL(s);
+
+            log.debug(uri + " --> " + url);
+            return url;
+        } catch (MalformedURLException ex) {
+            throw new RuntimeException("BUG: could not generate URL from uri " + s, ex);
+        }
+    }
+
+    protected String createURL(URI uri) throws IllegalArgumentException {
+
+        String newURL = "";
+
+        String schemeStr = uri.getScheme();
+
+        if (schemeStr.equals(SCHEME)) {
+            String path = uri.getSchemeSpecificPart();
+            if (path.isEmpty()) {
+                log.error(CANNOT_GET_URL + " Path portion of URI is empty: " + uri.toString());
+                throw new IllegalArgumentException(CANNOT_GET_URL + "Path is empty." + uri.toString());
+            }
+            newURL = MAST_BASE_ARTIFACT_URL + "/" + path;
+        } else {
+            log.error(CANNOT_GET_URL + " Invalid scheme (should be 'mast'): " + uri.toString());
+            throw new IllegalArgumentException(CANNOT_GET_URL + " Invalid scheme (should be 'mast'): " + uri.toString());
+        }
+
+        return newURL;
+    }
+
+    protected void validateScheme(URI uri) {
```

Comment:
this should be called validate() as it does more than check scheme

and never log.error and throw